### PR TITLE
[fix] Several minor fixes

### PR DESCRIPTION
**Pull request by bashimao**
See code mostly. Let's step through it together.

### CODE REVIEW

File: include/merlin_hashtable.cuh [hunk 1]
```diff
@@ -239,7 +239,7 @@ class HashTable {
    * the evict strategy. If false, it requires the metas follow the evict
    * strategy of table.
    */
-  void insert_or_assign(size_type n,
+  void insert_or_assign(size_t n,
```
Comment (rhdong) [inline include/merlin_hashtable.cuh:242]: Pls keep `size_type` here, follow the style of std::unordered_map

File: include/merlin_hashtable.cuh [hunk 2]
```diff
@@ -258,15 +258,15 @@ class HashTable {
...
     }
 
+    std::shared_lock<std::shared_timed_mutex> lock(mutex_, std::defer_lock);
+    if (!reach_max_capacity_) {
+      lock.lock();
+    }
+
     if (is_fast_mode()) {
-      const size_t block_size = 128;
+      constexpr size_t block_size = 128;
```
Comment (rhdong) [inline include/merlin_hashtable.cuh:267]: Wow, it's an existing bug, pls change it to option_.block_size, could you add new test case for none 128 block_size?

File: include/merlin_hashtable.cuh (hunk 1 above)
Comment (bashimao) [reply include/merlin_hashtable.cuh:242]: The usage of s`ize_type` is not consequent. This value is converted to size_t in the function.

File: include/merlin_hashtable.cuh (hunk 2 above)
Comment (bashimao) [reply include/merlin_hashtable.cuh:267]: Ok, that answers the question I had in Slack. Should I also replace all other instances where you do not use the `options_.block_size`?

File: include/merlin_hashtable.cuh (hunk 1 above)
Comment (bashimao) [reply include/merlin_hashtable.cuh:242]: You realize that the entire TableCore and all kernels don't have the size argument templated. Right?

Comment (bashimao) [reply include/merlin_hashtable.cuh:242]: Ok, Agreed!

File: include/merlin_hashtable.cuh (hunk 2 above)
Comment (bashimao) [reply include/merlin_hashtable.cuh]: Done as discussed on Slack!

**Review by rhdong — APPROVED** [review]
LGTM
